src/hooks/useUndernames/useUndernames.tsx

```diff
@@ -0,0 +1,260 @@
+import { useEffect, useState } from 'react';
+
+import { useArweaveCompositeProvider, useIsMobile } from '..';
+import {
+  CalendarTimeIcon,
+  ChevronUpIcon,
+  NotebookIcon,
+  PencilIcon,
+  TargetIcon,
+  TrashIcon,
+} from '../../components/icons/index';
+import CopyTextButton from '../../components/inputs/buttons/CopyTextButton/CopyTextButton';
+import {
+  ArweaveTransactionID,
+  PDNTContractJSON,
+  UndernameMetadata,
+} from '../../types';
+import eventEmitter from '../../utils/events';
+
+export function useUndernames(id: ArweaveTransactionID) {
+  const isMobile = useIsMobile();
+  const arweaveDataProvider = useArweaveCompositeProvider();
+  const [sortAscending, setSortOrder] = useState(true);
+  const [sortField, setSortField] = useState<keyof UndernameMetadata>('status');
+  const [selectedRow, setSelectedRow] = useState<UndernameMetadata>(); // eslint-disable-line
+  const [rows, setRows] = useState<UndernameMetadata[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [percent, setPercentLoaded] = useState<number | undefined>();
+
+  useEffect(() => {
+    if (id) {
+      setIsLoading(true);
+      fetchUndernameRows(id).finally(() => setIsLoading(false));
+    }
+  }, [id]);
+
+  function generateTableColumns(): any[] {
```

Comment:
Will these have the same shape? If so we can declare a type instead of using `any`. Not  blocker, we can have a tech_debt for it

Reply:
Actually rc and antd provide types for this, i added them.